File: src/main/java/grepp/NBE5_6_2_Team03/api/controller/mail/MailController.java
package grepp.NBE5_6_2_Team03.api.controller.mail;

import grepp.NBE5_6_2_Team03.api.controller.travelplan.dto.response.TravelPlanAdjustResponse;
import grepp.NBE5_6_2_Team03.domain.mail.service.MimeMailService;
import grepp.NBE5_6_2_Team03.domain.travelplan.service.TravelPlanQueryService;
import grepp.NBE5_6_2_Team03.domain.user.CustomUserDetails;
import java.util.Map;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/mail")
public class MailController {

    private final MimeMailService mimeMailService;
    private final TravelPlanQueryService travelPlanQueryService;

    @PostMapping("/send")
    public String sendSettlementMail(@RequestParam Long planId, @AuthenticationPrincipal CustomUserDetails customUser) {

        TravelPlanAdjustResponse response = travelPlanQueryService.getAdjustmentInfo(planId);
        Map<String,Object> settlementForm = Map.of("response",response);

        mimeMailService.sendSettlementMail(
            customUser.getUser().getEmail(),
            "정산 결과 안내",
            "settlement-summary",
            settlementForm
        );
        return "redirect:/users/home";
    }
}

File: tripin-mail/src/main/kotlin/com/grepp/mailservice/controller/MailController.kt
package com.grepp.mailservice.controller

import com.grepp.mailservice.dto.CodeMailRequest
import com.grepp.mailservice.dto.CodeMailResponse
import com.grepp.mailservice.dto.HtmlMailRequest
import com.grepp.mailservice.service.MailSendService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/mail")
class MailController(private val mailSendService: MailSendService) {

    @PostMapping("/send-code")
    fun sendSimpleMail(@RequestBody request: CodeMailRequest): ResponseEntity<CodeMailResponse> {
        val code = mailSendService.sendCodeMail(request.to, request.codeType)
        return ResponseEntity.ok(CodeMailResponse(request.to, code))
    }

    @PostMapping("/send-html")
    fun sendSettlementMail(@RequestBody request: HtmlMailRequest): ResponseEntity<Unit> {
        mailSendService.sendHtmlMail(request)
        return ResponseEntity.ok().build()
    }

}